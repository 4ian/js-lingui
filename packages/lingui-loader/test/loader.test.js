import compiler from "./compiler.js"

describe("lingui-loader", function() {
  let cwd

  beforeAll(function() {
    cwd = process.cwd()
    process.chdir(__dirname)
  })

  afterAll(function() {
    process.chdir(cwd)
  })

  it("should compile catalog", async () => {
    const stats = await compiler("./locale/en/messages.json")
    const output = stats.toJson().modules[0].source
    expect(output).toMatchSnapshot()
  })
})
