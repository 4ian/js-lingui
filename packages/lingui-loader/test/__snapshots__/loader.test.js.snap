// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lingui-loader should compile catalog 1`] = `"throw new Error(\\"Module build failed: TypeError: Cannot convert undefined or null to object\\\\n    at keys (<anonymous>)\\\\n    at Object.<anonymous>.exports.default (/Users/tricoder/Projects/lingui/js-lingui/packages/lingui-loader/src/index.js:19:48)\\");"`;
